blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpAggregatedMultiIntegrationTest.java

```diff
@@ -0,0 +1,110 @@
+        httpPort = Configuration.getInstance().getIntegerProperty(HttpConfig.HTTP_INGESTION_PORT);
+        manageShards.add(1); manageShards.add(5); manageShards.add(6);
+        context = spy(new ScheduleContext(System.currentTimeMillis(), manageShards));
+        HttpMetricsIngestionServer server = new HttpMetricsIngestionServer(context, new AstyanaxMetricsWriter());
+
+        httpIngestionService = new HttpIngestionService();
+        httpIngestionService.setMetricsIngestionServer(server);
+        httpIngestionService.startService(context, new AstyanaxMetricsWriter());
+
+        vendor = new HttpClientVendor();
+        client = vendor.getClient();
+    }
+
+    @Test
+    public void testHttpIngestionHappyCase() throws Exception {
+        HttpPost post = new HttpPost(getMetricsURI());
+        HttpEntity entity = new StringEntity(generateAggregatedJsonData(),
+                ContentType.APPLICATION_JSON);
+        post.setEntity(entity);
+        HttpResponse response = client.execute(post);
+        Assert.assertEquals(200, response.getStatusLine().getStatusCode());
+
+        final Locator locator = Locator.createLocatorFromPathComponents("5405532", "G200ms");
+        Points<GaugeRollup> points = AstyanaxReader.getInstance().getDataToRoll(GaugeRollup.class,
+                locator, new Range(1439231323000L, 1439231325000L), CassandraModel.getColumnFamily(GaugeRollup.class, Granularity.FULL));
+        Assert.assertEquals(1, points.getPoints().size());
+
+        final Locator locator1 = Locator.createLocatorFromPathComponents("5405577", "internal.bad_lines_seen");
+        Points<CounterRollup> points1 = AstyanaxReader.getInstance().getDataToRoll(CounterRollup.class,
+                locator1, new Range(1439231323000L, 1439231325000L), CassandraModel.getColumnFamily(CounterRollup.class, Granularity.FULL));
+        Assert.assertEquals(1, points1.getPoints().size());
+
+        EntityUtils.consume(response.getEntity()); // Releases connection apparently
+    }
+
+    @AfterClass
+    public static void shutdown() {
+        vendor.shutdown();
+    }
+
+    private URI getMetricsURI() throws URISyntaxException {
+        return getMetricsURIBuilder().build();
+    }
+
+    private URIBuilder getMetricsURIBuilder() throws URISyntaxException {
+        return new URIBuilder().setScheme("http").setHost("127.0.0.1")
+                .setPort(httpPort).setPath("/v2.0/acTEST/ingest/aggregated/multi");
+    }
+
+    public String generateAggregatedJsonData() throws Exception{
+        StringBuilder sb = new StringBuilder();
+        BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream("src/test/resources/sample_multi_bundle.json")));
+        String curLine = reader.readLine();
+        while (curLine != null) {
+            sb = sb.append(curLine);
+            curLine = reader.readLine();
+        }
+        String json = sb.toString();
+
+        return json;
+    }
+
+}
```

Comment:
In fact, what if, instead of 2 classes we had the same old class that now either accepts an object or a list of objects?